### ISSUE TITLE
Apply FullyQualifiedStrictTypesFixer in annotations

### ISSUE THREAD

**Issue by ostrolucky** [FriendsOfPHP/PHP-CS-Fixer#3655]
I would like to apply this fixer across annotations as well. Here's what I expect to fix

```php
<?php

class SomeClass
{
    /** @Foo\Bar */
    private $foo;
}
````

into

```php
<?php

use Foo\Bar;

class SomeClass
{
    /** @Bar */
    private $foo;
}
```

How do you think solution should look like? As far as I see, so far no fixer outside DoctrineAnnotation fixes annotations

**Comment by julienfalque** [comment]
:+1: though I think I would be better to have a dedicated fixer in the `DoctrineAnnotation` namespace.

**Comment by Slamdunk** [comment]
Already discussed in #3644

**Comment by ostrolucky** [comment]
Ah thanks for a link. Indeed, it would make sense to fix both cases (phpdoc param/return types and doctrine style classes prefixed with `@`)

**Comment by ostrolucky** [comment]
For all of those who are looking for same thing as I do, there is [SlevomatCodingStandard\Sniffs\Namespaces\ReferenceUsedNamesOnlySniff](https://github.com/slevomat/coding-standard#slevomatcodingstandardnamespacesreferenceusednamesonly-) which does exactly this

**Comment by ostrolucky** [comment]
For PHP-CS-Fixer, there is [Symplify\CodingStandard\Fixer\Import\ImportNamespacedNameFixer](https://github.com/Symplify/CodingStandard#types-should-not-be-referenced-via-a-fullypartially-qualified-name-but-via-a-use-statement)

**Comment by mvorisek** [comment]
I belive this is fixed by https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/5620

**Comment by Wirone** [comment]
@mvorisek no, #5620 was about phpDoc tags only (`@tag \F\Q\C\N`), not annotations (`@F\Q\C\N`).

**Comment by keradus** [comment]
btw, what is this `/** @MyClass */` annotation usage?

**Comment by mvorisek** [comment]
"attributes defined in phpdoc"

`/* @ORM\Column */` for example

https://packagist.org/packages/doctrine/annotations

**Comment by keradus** [comment]
I think Doctrine moved to attributes by now, if it's the only use case, maybe there is no big value of creating support for it.

**Comment by Wirone** [comment]
From [`doctrine/annotations`](https://github.com/doctrine/annotations):

> ⚠️ PHP 8 introduced attributes, which are a native replacement for annotations. As such, this library is considered feature complete, and should receive exclusively bugfixes and security fixes.

So it has limited support, but is not strictly deprecated. From our perspective it shouldn't be a priority IMHO, but if anyone is willing to contribute, then we can consider merging 😉.

**Comment by mvorisek** [comment]
with #7679 this can be easily implemented

**Comment by mvorisek** [comment]
!

**Comment by ostrolucky** [comment]
.
